File: templates/golang-grpc/internal/greeter/service.go
package greeter

import (
	"context"
	"strings"

	greeterv1 "example.com/__PROJECT_SLUG__/gen/greeter/v1"
	"google.golang.org/grpc/codes"
	"google.golang.org/grpc/status"
)

type Service struct {
	greeterv1.UnimplementedGreeterServiceServer
}

func (Service) SayHello(_ context.Context, request *greeterv1.SayHelloRequest) (*greeterv1.SayHelloResponse, error) {
	name := strings.TrimSpace(request.GetName())
	if name == "" {
		return nil, status.Error(codes.InvalidArgument, "name is required")
	}

	return &greeterv1.SayHelloResponse{Message: "Hello, " + name + "!"}, nil
}

File: templates/python-grpc/src/template_app/service.py
import grpc

from greeter.v1 import greeter_pb2, greeter_pb2_grpc


class GreeterService(greeter_pb2_grpc.GreeterServiceServicer):
    def SayHello(
        self,
        request: greeter_pb2.SayHelloRequest,
        context: grpc.ServicerContext,
    ) -> greeter_pb2.SayHelloResponse:
        name = request.name.strip()
        if not name:
            context.abort(grpc.StatusCode.INVALID_ARGUMENT, "name is required")
        return greeter_pb2.SayHelloResponse(message=f"Hello, {name}!")
